runtime/lua/vim/diagnostic.lua

```diff
@@ -1757,4 +1757,41 @@ function M.fromqflist(list)
   return diagnostics
 end
 
+--- Handle legacy diagnostic sign definitions
+--- These were deprecated in 0.10 and will be removed in 0.11
+do
+  local signs = vim.fn.sign_getdefined()
+  for _, sign in ipairs(signs) do
+    local severity = string.match(sign.name, '^DiagnosticSign(%w+)')
+    if M.severity[severity:upper()] then
+      local sev = M.severity[severity:upper()]
+      vim.deprecate(
```

Comment:
Soft deprecation does not include deprecation message. This should be gated behind a flag to only show for 0.11